### ISSUE TITLE
Procedures to package and deploy

### ISSUE THREAD

**Issue by ebekker** [angular/angular.io#776]
The angular.io docs do a great job of setting up the development environment and going through all the procedures of developing in Angular2.  But the one thing that seems to be missing is the best practices for packaging up an application and deploying it.

For example, would you want to deploy the node_modules folder and all its contents or is there a way to pack things up with just the compiled and needed dependencies.

**Comment by wardbell** [comment]
It's a great topic ... a huge topic.  Not sure when we'll get to it ... if ever. 

We'd welcome a literature/resource review that we could point to. Your contribution in this regard would be welcome.

**Comment by ebekker** [comment]
I would love to be able to contribute.  I've spent a bit of time searching around for references and I keep finding so many alternate "development bootstrap" examples using various different package managers (NPM, WebPack, JSPM) and unfortunately most of them are not current -- referencing older Angular2 procedures and example.

One resource that I was hopeful about was the [Angular2 Starter Kit](https://angularclass.github.io/angular2-webpack-starter/) but unfortunately, it doesn't appear to be working at least for me (but I'm still pecking at it).

The one issue I have is that the Angular2 [Quick Start](https://angular.io/docs/ts/latest/quickstart.html) sets you up with certain development bootstrap -- enough for a development mode environment, and then all the subsequent tutorials and guides build on that, but then there's no *conclusion* -- what do you do to deploy your work somewhere other than running in the [`light-server`](https://www.npmjs.com/package/lite-server).

**Comment by ebekker** [comment]
Found a good place to start on this topic with the [`angular-cli-github-pages` addon](https://github.com/IgorMinar/angular-cli-github-pages) (which I discovered on the fledgling angular-cli project (which I discovered thanks to [AiA](https://devchat.tv/adventures-in-angular/))).

Not sure if the tool will work for me as it is now, but at least I can see what it's trying to do under the hood and maybe extrapolate the general case.

Actually the ng-cli tool looks like it has some great promise in general for newbies like myself as well as seasoned devs.  I might make sense to *adjust* the Quick Start and related docs to make use of ng-cli once it becomes more stable.
